scenario/context.py

```diff
@@ -137,27 +344,33 @@ def run(
             charm will invoke when handling the Event.
         :arg pre_event: callback to be invoked right before emitting the event on the newly
             instantiated charm. Will receive the charm instance as only positional argument.
+            This argument is deprecated. Please use Context.event_manager instead.
```

Comment:
Could potentially issue `DeprecationWarning` if `pre_event` or `post_event` is set, or is that just annoying for users?

Reply:
I did add a logger.warning further below